src/include/storage/index/bwtree_index.h

```diff
@@ -40,13 +41,13 @@ class BwTreeIndex final : public Index {
     TERRIER_ASSERT(
         result,
         "non-unique index shouldn't fail to insert. If it did, something went wrong deep inside the BwTree itself.");
-
-    // Register an abort action with the txn context in case of rollback
-    txn->RegisterAbortAction([=]() {
-      const bool UNUSED_ATTRIBUTE result = bwtree_->Delete(index_key, location);
-      TERRIER_ASSERT(result, "Delete on the index failed.");
-    });
-
+    if (result) {
```

Comment:
Why are we adding this conditional when the assert immediately before it indicates it will always be true here?